webrender/src/platform/windows/font.rs

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use app_units::Au;
+use std::collections::HashMap;
+use webrender_traits::{FontKey, FontRenderMode, GlyphDimensions};
+
+use dwrote;
+
+pub struct FontContext {
+    // FIXME do I need to protect this in a mutex?
+    fonts: HashMap<FontKey, dwrote::Font>,
+    gdi_interop: dwrote::GdiInterop,
+    main_display_rendering_params: dwrote::RenderingParams,
+}
+
+pub struct RasterizedGlyph {
+    pub width: u32,
+    pub height: u32,
+    pub bytes: Vec<u8>,
+}
+
+impl FontContext {
+    pub fn new() -> FontContext {
+        FontContext {
+            fonts: HashMap::new(),
+            gdi_interop: dwrote::GdiInterop::create(),
+            main_display_rendering_params: dwrote::RenderingParams::create_for_primary_monitor(),
+        }
+    }
+
+    pub fn add_raw_font(&mut self, font_key: &FontKey, _: &[u8]) {
+        if self.fonts.contains_key(font_key) {
+            return
+        }
+
+        panic!("DWrite can't do raw fonts yet");
```

Comment:
Does this break a lot of websites? Should we consider making it non-fatal for now?